### ISSUE TITLE
Import new mods from mods folder

### ISSUE THREAD

**Issue by Nevrai** [killpowa/Twickt-Launcher#1421]
**Describe the bug**
I created a new instance and simply added a ‘mods’ folder, with 396 mods, to the instance folder. GDLauncher shows all the mods in the mods list, but when clicking ‘Add Mod’, most of the mods that are installed show up as not installed. It only recognizes some mods as installed whilst many of the others are shown as not installed, and if you click the download button on any of these mods that is already installed—but GDLauncher thinks isn’t—it adds a duplicate copy of the mod to the mods list. There’s still only one JAR within the mods folder, though, but now it shows up twice in the mods list, and the mod count goes up by one.

Additionally, updating doesn’t work with mods added to the ‘mods’ folder rather than installed within the launcher. Several of my mods have updates available, but there’s no update button, even if I click ‘Check for Updates’. However, if I uninstall the mod and install (an older version of) the same one within the launcher, _then_ it shows an update button.

I also tried importing a modpack .zip file from the CurseForge app, but the same issue occurs.

**To reproduce**
1. Create a new instance
2. Add mods to the instance’s ‘mods’ folder
3. Select the instance and click on ‘Add Mods’ on the instance’s ‘Mods’ tab
4. Search for a mod that’s already installed and notice that it appears as though it’s not installed
5. Notice that none of the installed mods have the update button, even if they are old versions

**Versions:**
* **GDLauncher version:** 1.1.27 (latest stable version)
* **OS:** Windows 11 Home, version 21H2

**Comment by Eskaan** [comment]
Yeah, it sees mods that are just added into the mods folder as external mods (overrides). You have to open the mod list for the instance and drag+drop them into there. Then it will check for each mod if it is a CF mod.

**Comment by Nevrai** [comment]
@Eskaan I see. Is it possible that you could add a simple button in GDLauncher that automatically fixes this issue for a particular instance, by recognizing all the mods in the instance’s ‘mods’ folder as non-external mods?

It seems a bit odd that this is a thing in the first place if it’s not a bug considering that you can’t update from the launcher either, which is inconvenient.

**Comment by Eskaan** [comment]
That's why I moved it to an enhancement, if someone wants to implement this they can do it.
